modules/common/memcached/memcached.go

```diff
@@ -0,0 +1,52 @@
+package memcached
+
+import (
+	"context"
+	"fmt"
+
+	memcachedv1 "github.com/openstack-k8s-operators/infra-operator/apis/memcached/v1beta1"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NewMemcached function will return a memcached v1beta1 instance of the CR. This can then be used
+// with controllerutil.CreateOrPatch() to create the resource in the cluster.
+func NewMemcached(name string, namespace string) *memcachedv1.Memcached {
+	return &memcachedv1.Memcached{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "memcached.openstack.org/v1beta1",
+			Kind:       "Memcached",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+}
+
+// GetMemcachedPodList gets a list of pods matching the labels provided and returns a slice of pod names.
+func GetMemcachedPodList(ctx context.Context, h *helper.Helper, memcachedName string, namespace string) ([]string, error) {
+	var podList []string
+	var memcachedPodList *corev1.PodList
+
+	// Define the label selector to filter on
+	labelSelector := metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"app":            "memcached",
+			"memcached/name": memcachedName,
+		},
+	}
+
+	memcachedPodList, err := h.GetKClient().CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{LabelSelector: metav1.FormatLabelSelector(&labelSelector)})
```

Comment:
I think we'd not need this any more as our operators are now cluster scoped. if we'd run the operator restricted mode with a multi namespace cache and you'd request pods from a not cached namespace you will run into this. https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/cache/multi_namespace_cache.go#L228

Reply:
Ah yeah, I see the GetPodListWithLabel. So I can re-use that. I guess I could use that from the operator directly, but it might be nice to have some common interface operators can use to get the memcached pods in the right format to be used. Since it's a little different than just getting the pod.Name, we also need to append the Headless service name as well. 

I can re-use the GetPodListWithLabel, it would only really change this line:
https://github.com/openstack-k8s-operators/lib-common/pull/218/commits/1f140f4455ea282c9872984c7ea319a264b5610a#diff-ac4c9ae55fd0190f9e0ca09ce5b6d4bebd8409eed043bc85b119dc29a82459b8R42

And as for why I used the client-go library compared to controller-util. I had previously used the services.go library to get services and noticed it was also using the client-go get() function rather than controller-util. So I just kept with the established standard that I found. There may be other examples of controller-util, I didn't put much extra effort into looking once I found that though, I just made the assumption we would always use that interface.